src/query/ee/src/stream/handler.rs

```diff
@@ -176,7 +176,11 @@ impl StreamHandler for RealStreamHandler {
             catalog
                 .drop_table_by_id(DropTableByIdReq {
                     if_exists: plan.if_exists,
-                    tenant,
+                    name_ident: TableNameIdent {
+                        tenant: tenant.clone(),
+                        db_name: plan.database.clone(),
+                        table_name: stream_name.clone(),
+                    },
                     tb_id: table.get_id(),
```

Comment:
What about passing in a `database_id: DatabaseId`?
Because what is actually required is a `database_id` by `drop_table_by_id`.

In order to decouple logic, 
I would always suggest moving logic(get db_meta) to a higher level(from `SchemaApi::drop_table_by_id()` to this this method `do_drop_stream()`.

Reply:
Another concern is that using `xxxName` internally still has a consistency issue, because `database-name` can be renamed by another transaction.